src/nanoarrow/common/inline_array.h

```diff
@@ -290,18 +293,27 @@ static inline ArrowErrorCode _ArrowArrayAppendEmptyInternal(struct ArrowArray* a
       case NANOARROW_BUFFER_TYPE_VARIADIC_SIZE:
       case NANOARROW_BUFFER_TYPE_VALIDITY:
         continue;
-      case NANOARROW_BUFFER_TYPE_DATA_OFFSET:
-        // Append the current value at the end of the offset buffer for each element
-        NANOARROW_RETURN_NOT_OK(ArrowBufferReserve(buffer, size_bytes * n));
-
-        for (int64_t j = 0; j < n; j++) {
-          ArrowBufferAppendUnsafe(buffer, buffer->data + size_bytes * (array->length + j),
-                                  size_bytes);
+      case NANOARROW_BUFFER_TYPE_SIZE:
+        NANOARROW_RETURN_NOT_OK(ArrowBufferAppendFill(buffer, 0, size_bytes * n));
+        continue;
+      case NANOARROW_BUFFER_TYPE_DATA_OFFSET: {
```

Comment:
The existing offset type is tightly bound to the idea that you have n+1 elements in the buffer to provide all open and closed intervals. The list views do not require that, so I've specialized this here. But we could alternatively make a new buffer enum member for offsets that aren't double ended (?)